src/Convolution.jl

```diff
@@ -0,0 +1,66 @@
+export Convolution
+
+"""
+	Convolution(G)
+
+Represents a convolution operator with function `G`. When applied to a function `f` it computes `∫_D G(x-y) f(y) dy for x ∈ D`. For now `D` is a `PeriodigSegment(a,b)` with `a<b∈R` and `space(G)` is either `Laurent(D)` or `Fourier(D)`.
+"""
+
+abstract type Convolution{S,T} <: Operator{T} end
+
+struct ConcreteConvolution{S<:Space,T} <: Convolution{S,T}
+    G::Fun{S,T}
+end
+
+function Convolution(G::Fun{S,T}) where {S,T}
+    @assert isfinite(arclength(domain(G)))
+    ConcreteConvolution(G)
+end
+ApproxFunBase.domain(C::ConcreteConvolution) = ApproxFunBase.domain(C.G)
+ApproxFunBase.domainspace(C::ConcreteConvolution) = ApproxFunBase.space(C.G)
+ApproxFunBase.rangespace(C::ConcreteConvolution) = ApproxFunBase.space(C.G)
+ApproxFunBase.bandwidths(C::ConcreteConvolution) = error("Please implement convolution bandwidths on "*string(space(C.G)))
+ApproxFunBase.getindex(C::ConcreteConvolution,k::Integer,j::Integer) = error("Please implement convolution getindex on "*string(space(C.G)))
+
+ApproxFunBase.bandwidths(C::ConcreteConvolution{Laurent{PeriodicSegment{R},T}}) where {R<:Real,T} = 0,0
+ApproxFunBase.bandwidths(C::ConcreteConvolution{Fourier{PeriodicSegment{R},T}}) where {R<:Real,T} = 1,1
+
+function ApproxFunBase.getindex(C::ConcreteConvolution{Laurent{PeriodicSegment{R},T1},T2},k::Integer,j::Integer) where {R<:Real,T1,T2}
```

Comment:
This should be `function getindex(C::`..., and not through `ApproxFunBase`

Reply:
ofc, I updated it